ecosystem/daos/jsondao.py

```diff
@@ -2,24 +2,93 @@
 from typing import Optional, List
 import os
 import json
+from pathlib import Path
+import shutil
+import toml
 
 from tinydb import TinyDB, Query
 
 from ecosystem.models import TestResult, StyleResult, CoverageResult, TestType
 from ecosystem.models.repository import Repository
 
 
+class EcosystemStorage:
+    """
+    Tell TinyDB how to read/write from files
+    (see https://tinydb.readthedocs.io/en/latest/extend.html)
+
+    File structure:
+
+    self.root
+    ├── members.json  # compiled file; don't edit manually
+    └── members
+        └── repo-name.toml
+
+    Database structure:
+
+    { tier: {  # e.g. Main, Community
+        index: repo  # `repo` is data from repo-name.toml
+    }}
+
+    """
+
+    def __init__(self, root_path):
+        self.root = Path(root_path)
+
+    def _url_to_path(self, url):
+        repo_name = url.strip("/").split("/")[-1]
+        return self.root / f"{repo_name}.toml"
+
+    def read(self):
+        """
+        Search for TOML files and add to DB
+        """
+        if not self.root.is_dir():
+            # For TinyDB initialization
+            return None
+
+        data = {}
+        for path in self.root.glob("*"):
+            repo = toml.load(path)
+            tier = repo["tier"]
+            if tier not in data:
+                data[tier] = {}
+            index = len(data[tier].keys())
+            data[tier][index] = repo
+
+        return data
+
+    def write(self, data):
+        """
+        Write TOML files, plus compiled JSON for qiskit.org
+        """
+        # Dump compiled JSON first
+        with open(self.root.with_suffix(".json"), "w") as file:
+            json.dump(data, file, indent=4)
+
+        # Erase existing human-readable files
+        if self.root.exists():
+            shutil.rmtree(self.root)
```

Comment:
Why won't this delete the JSON file? Also can you instead change line 77 to overwrite files rather than delete them all first? 

Although, I suppose you want to do cleanup of entries that no longer exist in the ecosystem? If so, please mention that in the comment because it's not obvious.

Reply:
The JSON file is `ecosystem/resources/members.json`, whereas `self.root` should be `ecosystem/resources/members`.

Happy to revise naming if you think it's confusing.

Reply:
Ah, yeah I recommend something less abstract. Probably `members_folder`. I would think `root` would be `ecosystem/resources`, and even there it's probably worth a more clear name like `parent_folder`, `root_folder`, or `resources_folder`.

Reply:
I'm not sure I agree with naming the folder `toml`. `members` is more descriptive. It's an implementation detail that we use TOML rather than YAML.

Reply:
That works for me too, is de92b80124b3b96964e8d5d296ef979f9b315e2e ok?